gammapy/background/make.py

```diff
@@ -0,0 +1,470 @@
+        #       in header there is EUNIT (TeV)!!!
+        #       hard coding the units for now !!!
+        ev_DETX = Angle(data_set['DETX'], 'degree')
+        ev_DETY = Angle(data_set['DETY'], 'degree')
+        ev_energy = Quantity(data_set['ENERGY'],
+                             data_set.meta['EUNIT'])
+        ev_cube_table = Table([ev_energy, ev_DETY, ev_DETX],
+                              names=('ENERGY', 'DETY', 'DETX'))
+        if DEBUG > 2:
+            print(ev_cube_table)
+
+        # fill events
+
+        # get correct data cube format for histogramdd
+        ev_cube_array = np.vstack([ev_cube_table['ENERGY'],
+                                   ev_cube_table['DETY'],
+                                   ev_cube_table['DETX']]).T
+
+        # fill data cube into histogramdd
+        ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
+                                                     [events_cube.energy_bins,
+                                                      events_cube.dety_bins,
+                                                      events_cube.detx_bins])
+        ev_cube_hist = Quantity(ev_cube_hist, '') # counts
+
+        # fill cube
+        events_cube.background += ev_cube_hist
+
+        # fill livetime for bins where E_max > E_thres
+        energy_max = events_cube.energy_bins[1:]
+        dummy_dety_max = np.zeros_like(events_cube.dety_bins[1:])
+        dummy_detx_max = np.zeros_like(events_cube.detx_bins[1:])
+        # define grid of max values (i.e. bin max values for each 3D bin)
+        energy_max, dummy_dety_max, dummy_detx_max = np.meshgrid(energy_max,
+                                                                 dummy_dety_max,
+                                                                 dummy_detx_max,
+                                                                 indexing='ij')
+        mask = energy_max > energy_threshold
+
+        # fill cube
+        livetime_cube.background += livetime*mask
+
+    return events_cube, livetime_cube
+
+
+def divide_bin_volume(cube):
+    """Divide by the bin volume.
+
+    Parameters
+    ----------
+    cube : `~gammapy.background.CubeBackgroundModel`
+        Cube containing the data to process.
+
+    Returns
+    -------
+    cube : `~gammapy.background.CubeBackgroundModel`
+        Cube divided by the bin volume.
+    """
+    delta_energy = cube.energy_bins[1:] - cube.energy_bins[:-1]
+    delta_y = cube.dety_bins[1:] - cube.dety_bins[:-1]
+    delta_x = cube.detx_bins[1:] - cube.detx_bins[:-1]
+    # define grid of deltas (i.e. bin widths for each 3D bin)
+    delta_energy, delta_y, delta_x = np.meshgrid(delta_energy, delta_y,
+                                                 delta_x, indexing='ij')
+    bin_volume = delta_energy.to('MeV')*(delta_y*delta_x).to('sr') # TODO: use TeV!!!
+    cube.background /= bin_volume
+
+    return cube
+
+
+def set_zero_level(cube):
```

Comment:
Same here ... why not make this a cube member function?
Is it because you are thinking about code re-use with other cube objects (like `SpectralCube`) that could also use this function?